Remotion/ObjectBinding/Web/UI/Controls/BocList.cs

```diff
@@ -3722,9 +3723,29 @@ IDropDownMenu IBocList.OptionsMenu
       }
     }
 
+    /// <inheritdoc />
+    public WebString? ListMenuHeading
+    {
+      get => _listMenu.Heading;
+      set => _listMenu.Heading = value;
+    }
+
+    /// <inheritdoc />
+    public HeadingLevel? ListMenuHeadingLevel
+    {
+      get => _listMenu.HeadingLevel;
+      set => _listMenu.HeadingLevel = value;
+    }
+
     IListMenu IBocList.ListMenu
     {
-      get { return _listMenu; }
+      get
+      {
+        if (ListMenuHeading is null or { IsEmpty: true })
+          ListMenuHeading = GetResourceManager().GetText(ResourceIdentifier.ListMenuHeading);
+
+        return _listMenu;
+      }
```

Comment:
For `ListMenuHeading` property, we need to use a dedicated backing field instead of directly piping the value into `_listMenu.Heading`, (see OptionsTitle). Otherwise, an empty value of the configured `ListMenuHeading` property will change once `ListMenu` getter has been used. This can result in persisting the wrong value in the aspx-markup and also introduce subtle bugs in application logic when programmatically setting the field.